scripts/post-install.sh

```diff
@@ -60,31 +58,44 @@ fi
 # Distribution-specific logic
 if [[ -f /etc/redhat-release ]]; then
     # RHEL-variant logic
-    which systemctl &>/dev/null
-    if [[ $? -eq 0 ]]; then
-	    install_systemd
+    if [[ "$(readlink /proc/1/exe)" == */systemd ]]; then
+        install_systemd
     else
-	    # Assuming sysv
-	    install_init
-	    install_chkconfig
+        # Assuming SysVinit
+        install_init
+        # Run update-rc.d or fallback to chkconfig if not available
+        if which update-rc.d &>/dev/null; then
+            install_update_rcd
+        else
+            install_chkconfig
+        fi
     fi
 elif [[ -f /etc/debian_version ]]; then
     # Debian/Ubuntu logic
-    which systemctl &>/dev/null
-    if [[ $? -eq 0 ]]; then
-	    install_systemd
-	    systemctl restart telegraf || echo "WARNING: systemd not running."
+    if [[ "$(readlink /proc/1/exe)" == */systemd ]]; then
+        install_systemd
+        systemctl restart telegraf || echo "WARNING: systemd not running."
     else
-	    # Assuming sysv
-	    install_init
-	    install_update_rcd
-	    invoke-rc.d telegraf restart
+	    # Assuming SysVinit
+        install_init
+        # Run update-rc.d or fallback to chkconfig if not available
+        if which update-rc.d &>/dev/null; then
+            install_update_rcd
+        else
+            install_chkconfig
```

Comment:
You're totally right, it's missing there. I'll rework the script and add support for it. Should be an easy quick fix.